core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/BatchWritableSink.java

```diff
@@ -49,7 +54,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     properties.remove(Properties.BatchReadableWritable.NAME);
     properties.remove(Properties.BatchReadableWritable.TYPE);
 
-    pipelineConfigurer.createDataset(datasetName, datasetType, DatasetProperties.builder().addAll(properties).build());
+    // null check for tests, as macro fields map and properties are null in test cases
```

Comment:
ok I see. That seems like just a bad test though. It's testing schema validation through the plugin instead of just testing the SchemaValidator.  Can we change these bad tests?  We should avoid adding logic strictly for tests.